kyo-sttp/shared/src/main/scala/kyo/internal/KyoSttpMonad.scala

```diff
@@ -55,9 +55,8 @@ class KyoSttpMonad extends MonadAsyncError[M]:
                     case Left(t)  => discard(p.complete(Result.panic(t)))
                     case Right(t) => discard(p.complete(Result.success(t)))
                 }
-            p.onComplete { r =>
-                if r.isPanic then
-                    canceller.cancel()
+            p.onInterrupt { _ =>
+                canceller.cancel()
```

Comment:
Use the new `onInterrupt` to cancel only in case of interruptions, not regular completions.